### ISSUE TITLE
Support for precomputed kernels

### ISSUE THREAD

**Issue by MichielStock** [willtebbutt/Stheno.jl#61]
Often you don't have (or need) a function to compute a kernel matrix on the fly, but you already have a precomputed Gram matrix. Example code how I solved this (because I needed a diffusion kernel):

```julia
struct Diffusion <: Kernel
    K
    function Diffusion(A::AbstractMatrix, β::Real=1.0; norm=false)
        L = laplacian(A)
        if norm
            L = norm_laplacian(L)
        end
        return new(exp(β * L))
    end
end

#Diffusion(E::Eigen, β::Real=1.0) = Diffusion(diffusion(E, β))

ew(k::Diffusion, x::AbstractVector{<:Int}) = [k.K[i,i] for i in x]
ew(k::Diffusion, x::AbstractVector{<:Int}, x′::AbstractVector{<:Int}) = [k.K[i,j] for (i,j) in zip(x, x′)]

pw(k::Diffusion, x::AbstractVector{<:Int}) = k.K[x,x]
pw(k::Diffusion, x::AbstractVector{<:Int}, x′::AbstractVector{<:Int}) = k.K[x,x′]
```

Maybe we should have `PrecomputedKernel` type to mediate this?

**Comment by willtebbutt** [comment]
Ahh I see. Yeah, a pre-computed kernel makes sense. Would be very happy to receive a PR involving a pre-computed kernel. I don't have a great intuition for what the best construction interface would be, but it might literally just be to pass in a matrix of numbers.

**Comment by willtebbutt** [comment]
Closed by #62 being merged. Please reopen if more to do.
